### PR TITLE
feat(auth): added jwt logic to the auth

### PR DESCRIPTION
**Pull request by Sanix-Darker**
## What
- implement the jwt on the auth
- removed the github token return by the jwt

## impact
In the future, to protect a route with the jwt token, we're going to use `app.IsAuthorized(targethandler)`

### CODE REVIEW

File: backend/app/router.go [hunk 1]
```diff
@@ -27,3 +32,54 @@ func GetRouter() *gin.Engine {
...
+		return "", err
+	}
+
+	return tokenString, nil
+}
+
+// IsAuthorized This function will check the authorization from a given token in Bearer
+func IsAuthorized(endpoint func(c *gin.Context)) gin.HandlerFunc {
+	fn := func(c *gin.Context) {
+
```
Comment (DipandaAser) [inline backend/app/router.go:59]: This new line is unecessary. this will fail during linting.
Delete it

**Comment by DipandaAser** [comment]
the linting check notice that the file are format with go fmt.
To format please run 
```shell
$> make fmt
```

**Review by DipandaAser — CHANGES_REQUESTED** [review]

File: backend/app/router.go (hunk 1 above)
Comment (Sanix-Darker) [reply backend/app/router.go:59]: Don't get it... what is unecessary ?
which lines ?

File: backend/app/router.go [hunk 2]
```diff
@@ -27,3 +32,54 @@ func GetRouter() *gin.Engine {
...
+	claims := token.Claims.(jwt.MapClaims)
+
+	claims["authorized"] = true
+	claims["client"] = customKey
+	claims["exp"] = time.Now().Add(time.Minute * 30).Unix()
+
+	tokenString, err := token.SignedString(jwtSigningKey)
+
+	if err != nil {
+		fmt.Errorf("Something Went Wrong: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (Sanix-Darker) [reply backend/app/router.go:49]: updated based on the return of the method : 
```
	if err != nil {
		return "", err
	}
```

**Pull request merged**